### PR TITLE
string dictionary start

### LINKED ISSUE #202: [Feature Request] - String Dictionary
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Is your feature request related to a problem, or an open issue? Describe the problem, and if applicable, reference the issue.

We want to pull out the strings that users read from the code and store them in a String Dictionary so when text is supposed to shown to the  user, the dictionary is referenced and the appropriate text is displayed. This way it is easier to customize the text that is displayed to the user and potentially adds the option to have support for multiple languages

## Describe the solution you'd like

Create a string dictionary that contain keys which stores text to be displayed and when text is to be show to the user, one of these keys is referenced instead

```typescript
export const stringDictionary: Record<string, string | (params: Record<string, string>) => string> = {}

/* TEAM FORMATION */
stringDictionary.teamRegistrationSuccess = (params) => `You have registered your team for sponsor judging. Our team will stop by during judging to see your hack. Best of luck and see you soon!\n**Team Name**: ${params.teamName}\n**Table Number**: ${params.tableNumber}\n**Project Description**: ${params.projectDescription}`
```

## Describe alternatives you've considered

An alternate was to just have string variables 

## Additional context

if we want to port this slack bot to discord then this feature is essential 
working with @oscargoddard

### PR DESCRIPTION
**Pull request by TheDivexz**
Co-Authored-By: Oscar Goddard <52873114+oscargoddard@users.noreply.github.com>

## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

creates a string dictionary

## Related Issues

Closes #202

### CODE REVIEW

**Comment by oscargoddard** [comment]
All the requirements are now completed. I am waiting for the reviews so that we can fix it with Dhruv.

File: src/slack/views/registerTeamSubmitted.ts [hunk 1]
```diff
@@ -112,23 +115,16 @@ Team Members: ${formattedTeamMembers.join(', ')}
...
-
-To help with resubmitting, here's the info you tried to submit:
-Team Name: ${teamName}
-TableNumber: ${tableNumber}
-Project Description: ${projectDescription}
-Team Members: ${formattedTeamMembers.join(', ')}
-
-Here's what went wrong, it may be helpful (but probably not):
-\`\`\`
-${JSON.stringify(err, null, 2)}
```
Comment (oscargoddard) [inline src/slack/views/registerTeamSubmitted.ts:129]: I am not sure if we are covering this line when passing the string from dictionary.

Comment (SpencerKaiser) [reply src/slack/views/registerTeamSubmitted.ts:129]: I’m not following your question

Comment (oscargoddard) [reply src/slack/views/registerTeamSubmitted.ts:129]: I couldnt make the team registration fail. I am not sure if this JSON is available to user when the registration fails.

**Comment by oscargoddard** [comment]
Finalized all the changes requested at Zoom Meeting. Fixed all the linting errors. Waiting for review.

File: src/slack/views/registerTeamSubmitted.ts (hunk 1 above)
Comment (oscargoddard) [reply src/slack/views/registerTeamSubmitted.ts:129]: There was a problem me not being able to fail the team registration. I solved it.

**Comment by oscargoddard** [comment]
Finalized all the changes requested at Zoom Meeting. Fixed all the linting errors. Waiting for review.

**Review by oscargoddard — APPROVED** [review]

**Review by johnkahn — APPROVED** [review]

**Pull request merged**